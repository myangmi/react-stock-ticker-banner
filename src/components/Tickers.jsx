import React from 'react';
import PropTypes from 'prop-types';

import './Tickers.scss';

const propTypes = {
  quoteData: PropTypes.object,
};

const defaultProps = {
  quoteData: {},
};

const Tickers = ({ quoteData }) => {
  const content = (
    <>
      <ul className='Tickers'>
        {
          Object.keys(quoteData).map((ticker, i) => (
            <li key={i}>{`${ticker}`}: ${`${quoteData[ticker].price}`}</li>
          ))
        }
        {
          Object.keys(quoteData).map((ticker, i) => (
            <li key={i}>{`${ticker}`}:  ${`${quoteData[ticker].price}`}</li>
          ))
        }
      </ul>
    </>
  );

  return (
    content
  );
};

Tickers.defaultProps = defaultProps;
Tickers.propTypes = propTypes;
export default Tickers;
