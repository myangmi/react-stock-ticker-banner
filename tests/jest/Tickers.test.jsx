import React from 'react';
import { shallow } from 'enzyme';

import Tickers from '../../src/components/Tickers.jsx';
import mockBatchIEX from '../mocks/mockBatchIEX.json';

describe('Tickers', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Tickers />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render mocked data properly', () => {
    const wrapper = shallow(<Tickers quoteData={mockBatchIEX}/>);

    expect(wrapper.find('li').at('0').text()).toEqual('AMD: $150.48');
    expect(wrapper.find('li').at('1').text()).toEqual('MSFT: $327.77');
    expect(wrapper.find('li').at('2').text()).toEqual('AMZN: $3461.5');
    expect(wrapper.find('li').at('3').text()).toEqual('INTC: $49.27');
    expect(wrapper.find('li').at('4').text()).toEqual('TSM: $121.35');
    expect(wrapper.find('li').at('5').text()).toEqual('AAPL: $169.51');
    expect(wrapper.find('li').at('6').text()).toEqual('FB: $307.57');
    expect(wrapper.find('li').at('7').text()).toEqual('NFLX: $613.96');

    expect(wrapper).toMatchSnapshot();
  });
});
