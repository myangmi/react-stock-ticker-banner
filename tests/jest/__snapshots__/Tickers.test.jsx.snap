// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tickers should render mocked data properly 1`] = `
<Fragment>
  <ul
    className="Tickers"
  >
    <li
      key="0"
    >
      AMD
      : $
      150.48
    </li>
    <li
      key="1"
    >
      MSFT
      : $
      327.77
    </li>
    <li
      key="2"
    >
      AMZN
      : $
      3461.5
    </li>
    <li
      key="3"
    >
      INTC
      : $
      49.27
    </li>
    <li
      key="4"
    >
      TSM
      : $
      121.35
    </li>
    <li
      key="5"
    >
      AAPL
      : $
      169.51
    </li>
    <li
      key="6"
    >
      FB
      : $
      307.57
    </li>
    <li
      key="7"
    >
      NFLX
      : $
      613.96
    </li>
    <li
      key="0"
    >
      AMD
      :  $
      150.48
    </li>
    <li
      key="1"
    >
      MSFT
      :  $
      327.77
    </li>
    <li
      key="2"
    >
      AMZN
      :  $
      3461.5
    </li>
    <li
      key="3"
    >
      INTC
      :  $
      49.27
    </li>
    <li
      key="4"
    >
      TSM
      :  $
      121.35
    </li>
    <li
      key="5"
    >
      AAPL
      :  $
      169.51
    </li>
    <li
      key="6"
    >
      FB
      :  $
      307.57
    </li>
    <li
      key="7"
    >
      NFLX
      :  $
      613.96
    </li>
  </ul>
</Fragment>
`;

exports[`Tickers should render properly 1`] = `
<Fragment>
  <ul
    className="Tickers"
  />
</Fragment>
`;
